src/connectors/impls/gcs/streamer.rs

```diff
@@ -167,19 +167,27 @@ async fn execute_http_call(
     api_client: &mut impl ApiClient,
     request: HttpTaskRequest,
 ) -> Result<()> {
+    let send_ack = matches!(request.command, HttpTaskCommand::FinishUpload { .. });
     let result = api_client
         .handle_http_command(done_until.clone(), &config.url, request.command)
         .await;
 
     match result {
         Ok(_) => {
-            if let Some(contraflow_data) = request.contraflow_data {
-                reply_tx
-                    .send(AsyncSinkReply::Ack(
-                        contraflow_data,
-                        nanotime() - request.start,
-                    ))
-                    .await?;
+            if send_ack {
+                // we only ack the event upon the finished upload, not before
+                // as we cannot be sure yet that it was actually uploaded
+                if let Some(contraflow_data) = request.contraflow_data {
+                    reply_tx
+                        .send(AsyncSinkReply::Ack(
+                            contraflow_data,
+                            0, // duration of event handling is a little bit meaningless for this sink
+                               // the actual duration will be very long due to the accumulating nature of this sink
+                               // it will vary a lot when we do an actual upload vs. when we only accumulate
+                               // ergo we don't track it here
+                        ))
+                        .await?;
+                }
             }
         }
         Err(e) => {
```

Comment:
In case of logs this just means a log line is missing. This is not necessary corrupted.

Reply:
As I mentioned in the main comment, those are two very different use cases, in the case of logs, we would also not want to re-send all the logs, which I think we'd do right now so we will have to specialize on that front in some way, but if we don't fail the entire upload in case of an error and still re-send it all via CB it's not in the sweetspot for either logs or objects

Reply:
If events are not transactional, we will not send any ack or fail. This is exactly the log use case. If shit fails, we don't care.

We could add a setting that invalidates an upload if any part of it fails, so it is either the whole upload or nothing. Together with transactional events, this will give the binary behaviour of either the upload is completely there or not at all.

Reply:
I've a hard time coming up with a use case of "we have transactional data in and replaying them isn't an issue". I can totally see use cases for "we have transactional data and need it correct" and "we have non-transactional data and YOLO.


let me give an example, if we got  events 1-6, 3 fails requests a replay but 4 and 5 are already in the fight we would get a file like this: `[1, 2, 4, 5, 3, 4, 5]` and I can't see a place where this would be correct behavior for a file.

Reply:
We resolved this in a discussion on discord. We now have a consistent and yolo mode. Docs are being updated.